File: jsParser/parser/Type_specifierVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Name = require('../domain/Name')
const Type_specifier = require('../domain/Type_specifier')

const NameVisitor = require('./NameVisitor')

class Type_specifierVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitType_specifier (ctx) {
    let name = null

    if (ctx.name()) {
      const nameVisitor = new NameVisitor.NameVisitor()
      name = nameVisitor.visitName(ctx.name())
    }

    return new Type_specifier.Type_specifier(name)
  }
};

Type_specifierVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitType_specifier = this.visitType_specifier
exports.Type_specifierVisitor = Type_specifierVisitor

File: java/parser/src/main/java/gov/lbl/parser/parser/Type_specifierVisitor.java
package gov.lbl.parser.parser;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Name;
import gov.lbl.parser.domain.Type_specifier;

public class Type_specifierVisitor  extends modelicaBaseVisitor<Type_specifier> {
    @Override
    public Type_specifier visitType_specifier(modelicaParser.Type_specifierContext ctx) {
        NameVisitor nameVisitor = new NameVisitor();
        Name name = ctx.name() == null ? null : ctx.name().accept(nameVisitor);
        return new Type_specifier(name);
    }
}